pravega/pravega
# Issue 6498:Hardcoded port 8000 in standalone mode

Signed-off-by: Shwetha N <shwetha.n1@dell.com>

**Change log description**  
In Standalone mode default JDWP(8050 as Java Debug port) can be overridden through environment variable.

**Purpose of the change**  
Fixing #6498 

**What the code does**  
JDWP(Java debug port) is used only if user wants to debug the code . 
If user wants to troubleshoot running java application remotely and wants to use other port(other than default 8050), then set it through environment variable before starting Pravega in standalone mode(either it can be from Source code or from installation package).
User can set the environment variable using below command.
export JDWP_PORT=8888
In above command **8888** , 8888 can be changed to any other value, ex: 8020 and 0 - 65535 are allowed port range

**How to verify it**  
1. Start Pravega standalone from installation package
        A. Start Pravega without JDWP port enabled : $ pravega-/bin/pravega-standalone
        B. Start Pravega with JDWP port enabled:
             $ export JDWP_PORT=8888
             $ pravega-/bin/pravega-standalone

2. Start Pravega standalone from Source code
        A. Start Pravega without JDWP port enabled : $ ./gradlew startStandalone
        B. Start Pravega with JDWP port enabled:
                $ export JDWP_PORT=8888
                $ ./gradlew startStandalone

3. To reset debug arguments
        $ export JDWP_PORT=

4. All tests should pass
